### PR TITLE
Improved camera operations

### PR DESCRIPTION
**Pull request by jweisz**
**Scope of Changes**

- Added documentation of `tj.recognizeObjectsInPhoto()`, `tj.recognizeTextInPhoto()`
- Removed ununsed method `tj._captureImage()`
- Added `time` property to node-raspistill camera options (and swizzling it in just in case, if for some reason, the version of node-raspistill doesn't support the `time` property) in order to speed up image capture from the pi camera
- Requiring node-raspistill greater or equal to 0.0.11 to support the `time` property
- Setting some additional camera options every time we take a photo, just in case the user changed those options in between taking photos (e.g. width, height, vflip, hflip)
- Bumped version to 1.2.1

### CODE REVIEW

**Comment by victordibia** [comment]
Can you confirm the `t=1` flag solves the 6 seconds delay issue?

File: lib/tjbot.js [hunk 1]
```diff
@@ -199,8 +199,23 @@ TJBot.prototype._setupCamera = function() {
         encoding: 'jpg',
         outputDir: './',
         verticalFlip: this.configuration.see.camera.verticalFlip,
-        horizontalFlip: this.configuration.see.camera.horizontalFlip
+        horizontalFlip: this.configuration.see.camera.horizontalFlip,
+        time: 1
```
Comment (victordibia) [inline lib/tjbot.js]: `timeout` instead?

https://www.raspberrypi.org/wp-content/uploads/2013/07/RaspiCam-Documentation.pdf

File: lib/tjbot.js [hunk 2]
```diff
@@ -199,8 +199,23 @@ TJBot.prototype._setupCamera = function() {
...
     });
+    
+    // versions of node-raspistill < 0.0.11 don't have the `time` option, so
+    // force it in if we don't find it
+    if (!this._camera.options.hasOwnProperty('time')) {
```
Comment (victordibia) [inline lib/tjbot.js]: `timeout`? see above.

File: lib/tjbot.js (hunk 1 above)
Comment (victordibia) [inline lib/tjbot.js]: Would be great if you can confirm the `t=1` flag solves the earlier 6 seconds problem I mentioned earlier. 
Note: benchmarchs using spawn takes about 1.2 seconds.

Comment (jweisz) [inline lib/tjbot.js]: i believe it's `time`, as per node-raspistill's options:
https://github.com/dlukanin/node-raspistill/blob/master/src/lib/camera/abstract.ts

```
/**
     * Map "option and option value -> raspistil exec arg"
     * @type {any}
     */
    protected optionsMap: {[key: string]: string} = {
        verticalFlip: '-vf',
        horizontalFlip: '-hf',
        noPreview: '-n',
        encoding: '-e',
        width: '-w',
        height: '-h',
        timelapse: '-tl',
        time: '-t'
    };
```

Comment (jweisz) [inline lib/tjbot.js]: I can confirm the code path that uses the swizzled method works, it takes like a second and i can see that the `-t 1` option is being included in the call to `raspistill`.  i cannot yet confirm whether the inclusion of `time: 1` in the camera options also causes `-t 1` to be included in the call to `raspistill` (using node-raspistill 0.0.11), due to some trouble i had with npm install including an older version of the library.

File: lib/tjbot.js (hunk 2 above)
Comment (jweisz) [inline lib/tjbot.js]: `time`, not `timeout`, it's a property on node-raspistill's `AbstractCamera` object

File: lib/tjbot.js (hunk 1 above)
Comment (victordibia) [inline lib/tjbot.js]: I'll test this out later.

**Pull request merged**